src/database/datatools/datadescription.py

```
@@ -0,0 +1,155 @@
+        _format_field("Standard Material Ref ID", data[3]),
+        _format_field("No-Mass Material Ref ID", data[4])
+    ]
+    content = " | ".join([f for f in fields if f])
+    de = 'The all_materials index data contains the parameters required in EnergyPlus. The attribute id corresponds to the id of this material in the corresponding data table. The ID represents its ID in the all_materials table. '
+    return f"This is all_materials index data in our EnergyPlus database. {de} EnergyPlus Material Cross-Reference: {content}"
+
+def _update_description(db_path, table_name, data, gen_func):
+    conn = sqlite3.connect(db_path)
+    cursor = conn.cursor()
+    description = gen_func(data)
+    sql = "UPDATE " + table_name + " SET description = ? WHERE id = ?"
```

Ruff (0.15.0)

[error] Possible SQL injection vector through string-based query construction

(S608)